### PR TITLE
Use CSSLintBear

### PR DESCRIPTION
**Pull request by KVGarg**
The commit adds a new section in .coafile
which will be responsible for linting css files
and, check if there are any error or not.

### CODE REVIEW

File: resources/css/style.css [hunk 1]
```diff
@@ -1,4 +1,4 @@
-.all-filters-option {
+.all-filters-option {  /* Ignore CSSLintBear */
```
Comment (KVGarg) [inline resources/css/style.css]: I tried to alter positions of `-webkit*` css styles, but it wasn't affecting the error. It was still getting raised.

File: resources/css/style.css [hunk 2]
```diff
@@ -71,7 +71,7 @@
 .filter-projects-inputs {
   display: flex;
   flex-flow: row wrap;
-  justify-content: space-evenly;
+  justify-content: space-evenly;  /* Ignore CSSLintBear */
```
Comment (KVGarg) [inline resources/css/style.css]: csslint doesn't have a supporting rule for this. It doesn't accepts `space-evenly` 
Already raised an issue related to this in upstream

File: resources/css/style.css [hunk 3]
```diff
@@ -91,14 +91,14 @@
...
-  padding-top: 20px;
   padding-bottom: 10px;
+  padding-top: 20px;
   width: 100%;
 }
 .footer-icons {
   display: flex;
   flex-wrap: wrap;
-  justify-content: space-evenly;
+  justify-content: space-evenly;  /* Ignore CSSLintBear */
```
Comment (KVGarg) [inline resources/css/style.css]: same as above

File: resources/css/style.css [hunk 4]
```diff
@@ -174,7 +174,7 @@ nav {
   text-align: center;
 }
 .report .card {
-  word-break: break-word;
+  word-break: break-word;  /* Ignore CSSLintBear */
```
Comment (KVGarg) [inline resources/css/style.css]: same as above! Doesn't have a support for `break-word` attr value

File: resources/css/style.css [hunk 5]
```diff
@@ -198,20 +198,14 @@ nav {
...
-  width: 85%;
-  min-width: 340px;
   margin-top: 0;
+  min-width: 340px;
+  width: 85%;
 }
 .sha256sum_hash {
   display: flex;
-  justify-content: space-evenly;
+  justify-content: space-evenly;  /* Ignore CSSLintBear */
```
Comment (KVGarg) [inline resources/css/style.css]: same as above

**Comment by abhishalya** [comment]
@KVGarg Can you rebase this PR, it would good to merge this one first.

**Comment by abhishalya** [comment]
@KVGarg The CI should pass now, seems like there's some other issue.

**Comment by KVGarg** [comment]
> @KVGarg The CI should pass now, seems like there's some other issue.

Yeah! the ruby bundler isn't found

**Comment by KVGarg** [comment]
@abhishalya ready to be merged

File: .travis.yml [hunk 6]
```diff
@@ -35,6 +40,7 @@ jobs:
         - curl "${GRAVIS}.install-jdk-travis.sh"
           --output ~/.install-jdk-travis.sh
         - source ~/.install-jdk-travis.sh
+        - npm install --no-save
```
Comment (abhishalya) [inline .travis.yml:43]: Why are we doing this?

Comment (abhishalya) [reply .travis.yml:43]: I would rather avoid committing package.json just for csslint.

Comment (KVGarg) [reply .travis.yml:43]: But, to get this css lint work we need this package.json and run this commans. I found this from an existing coala repository https://github.com/coala/coala-html/blob/master/package.json

Comment (abhishalya) [reply .travis.yml:43]: Yes, but we don't need a package.json for this repo. So, better just add `npm install csslint` in travis.yml to get the csslint executable.

Comment (abhishalya) [reply .travis.yml:43]: Also by that I don't think we'd need the manual bundler install. Will need to check this though, so try removing that bundler change and push with just csslint install. If that still breaks, we can add the custom bundler install command again.

Comment (KVGarg) [reply .travis.yml]: Removed last commit and package.json too. Build failed due to `bundler` :disappointed:

**Comment by abhishalya** [comment]
@KVGarg Hmm, not sure what is happening with travis since it should use bundler 1.x as in the Gemfile.lock (as it does for the existing master build). Okay, for now we had enough of this. Let's just manually install bundler and fix this. If we find a better solution, a PR would be fine imo.

File: .travis.yml [hunk 7]
```diff
@@ -3,7 +3,16 @@ services: docker
 language: ruby
 rvm:
   - 2.5.1
-cache: bundler
+cache:
+  bundler: true
+  directories:
+    - node_modules
```
Comment (abhishalya) [inline .travis.yml:9]: Don't need this now?

File: .travis.yml [hunk 8]
```diff
@@ -3,7 +3,16 @@ services: docker
...
+
+env:
+  global:
+    - PATH="$PATH:$TRAVIS_BUILD_DIR/node_modules/.bin"
```
Comment (abhishalya) [inline .travis.yml:13]: This as well

File: .travis.yml [hunk 9]
```diff
@@ -12,13 +21,9 @@ cache: bundler
...
   script: []
 
-env:
-  global:
-    - GRAVIS="https://raw.githubusercontent.com/DanySK/Gravis-CI/master/"
-    - JDK="adopt@1.8.0-212"
-
 jobs:
   include:
+    - stage: test
```
Comment (abhishalya) [inline .travis.yml:26]: ?

File: .travis.yml (hunk 8 above)
Comment (KVGarg) [reply .travis.yml:13]: Travis couldn't locate the path of node modules, so there was need to specify it.

Comment (abhishalya) [reply .travis.yml:13]: No, use `npm install -g csslint` then, it should work fine.

**Comment by abhishalya** [comment]
@KVGarg Check https://github.com/abhishalya/projects/tree/upstream/%23761 the config there works.

File: .travis.yml [hunk 10]
```diff
@@ -47,6 +48,7 @@ jobs:
 before_install:
   # Speeds installation of html-proofer
   - export NOKOGIRI_USE_SYSTEM_LIBRARIES=true
+  = gem install bundler -v '< 2.0'
```
Comment (abhishalya) [inline .travis.yml:51]: No, not required.

Comment (abhishalya) [reply .travis.yml:51]: Check the config I mentioned

Comment (KVGarg) [reply .travis.yml:51]: Can you list me out what all changes are required?

Comment (abhishalya) [reply .travis.yml:51]: Just a single change, related to installing of csslint.

Comment (KVGarg) [reply .travis.yml:51]: but wtb bundler then? won't it fail?

Comment (abhishalya) [reply .travis.yml:51]: You can probably just copy the entire `.travis.yml` file from my fork above.

Comment (abhishalya) [reply .travis.yml:51]: No, it won't.

File: .travis.yml [hunk 11]
```diff
@@ -60,3 +61,4 @@ notifications:
 
 # Route your build to the container-based infrastructure for a faster build
 sudo: false
+
```
Comment (abhishalya) [inline .travis.yml:64]: Redundant. We just need single blank line at the end.

**Comment by KVGarg** [comment]
Maybe the bundler was not getting installed in stage test that I declared :rofl: I don't know why I declared that new stage :thinking:

File: .travis.yml (hunk 11 above)
Comment (KVGarg) [reply .travis.yml]: Yeah! removed that

**Comment by KVGarg** [comment]
Finally! A green check from travis

**Pull request merged**